### PR TITLE
Refactor CRIUConfigurator

### PR DESCRIPTION
**Pull request by tajila**
Refactor CRIUConfigurator

Move CRIUConfigurator to a new package that is only exported to
openj9.criu

Signed-off-by: Tobi Ajila <atobia@ca.ibm.com>

### CODE REVIEW

**Comment by tajila** [comment]
Jenkins compile xlinuxcriu jdk11

File: jcl/src/java.base/share/classes/openj9/internal/criu/security/CRIUConfigurator.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import sun.security.jca.Providers;
+
+/**
+ * Configures the security providers when in CRIU mode.
+ */
+public final class CRIUConfigurator {
+    private static Properties systemProps;
+    /** Stores the old security providers (position, name). */
+    private static final HashMap<String, String> oldProviders = new HashMap<>();
+    /** Tracing for CRIUSEC. */
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/openj9/internal/criu/security/CRIUConfigurator.java:42]: Please indent with tabs for consistency with this repo.

File: jcl/src/java.base/share/classes/module-info.java.extra [hunk 2]
```diff
@@ -63,10 +63,12 @@ exports com.ibm.gpu.spi to
 /*[IF CRIU_SUPPORT]*/
 exports openj9.internal.criu to
     openj9.criu;
+exports openj9.internal.security to
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/module-info.java.extra:66]: This export doesn't match any package; I think you meant to export `openj9.internal.criu.security`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: jcl/src/java.base/share/classes/openj9/internal/criu/security/CRIUConfigurator.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+	/**
+	 * Removes the CRIU security provider and adds the usual security providers
+	 * back.
+	 */
+	public static void setCRIURestoreMode() {
+		Security.removeProvider("CRIUSEC");
+		// Note that CRIUSEC was set as security.provider.1 in the method
+		// setCRIUSecMode,
+		// which is called before this method.
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/openj9/internal/criu/security/CRIUConfigurator.java:79]: Please merge these comment lines.

**Comment by tajila** [comment]
I've addressed the compilation and indentation issues

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
jenkins compile osx,xlinuxcriu jdk11

**Comment by keithc-ca** [comment]
jenkins compile xlinuxcriu jdk11 depends ibmruntimes/openj9-openjdk-jdk11#523

**Comment by keithc-ca** [comment]
For the record, both builds were successful:
* https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/2109
* https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/2110

**Pull request merged**